libraries/resource_providers/api500/c7000/volume_template_provider.rb

```diff
@@ -0,0 +1,51 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API500
+    module C7000
+      # VolumeTemplate API500 C7000 provider
+      class VolumeTemplateProvider < API300::C7000::VolumeTemplateProvider
+        # Loads the VolumeTemplate with all the external resources (if needed)
+        def load_resource_with_associated_resources
+          @item.set_root_template(load_root_template)
+          @item.set_default_value('storagePool', load_storage_pool)
+          @item.set_default_value('snapshotPool', load_snapshot_pool) if @new_resource.snapshot_pool
+        end
+
+        # Loads the Snapshot Pool
+        # @return [resource_named(:StoragePool)] The Snapshot Pool loaded
+        def load_snapshot_pool
+          load_storage_system unless @storage_system
+          @snapshot_pool = resource_named(:StoragePool).find_by(@item.client, name: @new_resource.snapshot_pool, storageSystemUri: @storage_system['uri']).first
```

Comment:
It is not necessary to be a global variable, because it is used only in this method.